source/h2_decoder.c

```diff
@@ -1263,6 +1282,30 @@ static int s_process_header_field(struct aws_h2_decoder *decoder, const struct a
 
         /* #TODO Validate characters used in header_field->value */
 
+        if (name_enum == AWS_HTTP_HEADER_COOKIE) {
+            /* for a header cookie, we will not fire callback until we concatenate them all, let's store it at the
+             * buffer */
+            if (header_field->compression > current_block->cookie_header_compression_type) {
+                current_block->cookie_header_compression_type = header_field->compression;
+            }
+
+            if (current_block->cookies.len) {
+                /* add a delimiter */
+                struct aws_byte_cursor delimiter = aws_byte_cursor_from_c_str("; ");
+                if (aws_byte_buf_append_dynamic(&current_block->cookies, &delimiter)) {
+                    DECODER_LOG(ERROR, decoder, "Store cookie delimiter to buffer failed");
+                    return AWS_OP_ERR;
+                }
+            }
+            if (aws_byte_buf_append_dynamic(&current_block->cookies, &header_field->value)) {
+                DECODER_LOG(ERROR, decoder, "Store the current cookie value to buffer failed");
+                DECODER_LOGF(
+                    DEBUG, decoder, "Value of cookie is '" PRInSTR "'", AWS_BYTE_CURSOR_PRI(header_field->value));
```

Comment:
super trivial: don't bother logging errors that can only fail due to out-of-memory. They're so incredibly unlikely it's barely worth the effort of writing the commands. And if a header actually runs us out of memory, probably not a good idea to try and log its value